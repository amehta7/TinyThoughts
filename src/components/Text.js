import React from 'react'
import PropTypes from 'prop-types'

const Text = ({ label, onUpdate, value, maxLength, hasError }) => {
  return (
    <div className={hasError ? 'text-box hasError' : 'text-box'}>
      <span className='label'>{label}</span>
      <textarea
        value={value}
        maxLength={maxLength}
        type='textarea'
        onChange={onUpdate}
      />
    </div>
  )
}

Text.propTypes = {
  type: PropTypes.string,
}

export default Text
