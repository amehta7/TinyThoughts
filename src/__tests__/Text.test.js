import React from "react";
import { shallow } from "enzyme";
import Text from "../components/Text";

describe("Text component", () => {
  it("Renders a controlled component", () => {
    const onSetFn = jest.fn();
    const wrapper = shallow(
      <Text
        label="Your comments"
        onUpdate={onSetFn}
        value={"This is a text box"}
        hasError={false}
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper
      .find("textarea")
      .simulate("change", { target: { value: "And here we go!" } });
    expect(onSetFn).toHaveBeenCalledTimes(1);
  });

  it("Renders a red border when hasError is set to true", () => {
    const wrapper = shallow(
      <Text
        label="Your comments"
        onUpdate={() => {}}
        value={"This is a text box"}
        hasError={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
