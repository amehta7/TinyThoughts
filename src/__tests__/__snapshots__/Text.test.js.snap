// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Renders a controlled component 1`] = `
<div
  className="text-box"
>
  <span
    className="label"
  >
    Your comments
  </span>
  <textarea
    maxLength={200}
    onChange={[Function]}
    value="This is a text box"
  />
</div>
`;

exports[`Text component Renders a red border when hasError is set to true 1`] = `
<div
  className="text-box hasError"
>
  <span
    className="label"
  >
    Your comments
  </span>
  <textarea
    maxLength={200}
    onChange={[Function]}
    value="This is a text box"
  />
</div>
`;
